budgetportal/models.py

```diff
@@ -231,6 +233,36 @@ def get_contributed_datasets(self):
                     datasets[package['name']] = dataset
         return datasets.values()
 
+    def get_budget_totals(self, year):
+        """
+        get the budget totals for all the department programmes
+        """
+        resource = {
+            '2015': 'estimates-of-national-expenditure-south-africa-2015-16/',
+            '2016': 'estimates-of-national-expenditure-south-africa-2016-17/',
+            '2017': 'estimates-of-national-expenditure-south-africa-2017-18/'
+        }
+        params = {
+            'cut': ('date_2.financial_year:{}|'
+                    'administrative_classification_2.department:"{}"')
+            .format(year.slug[:4], self.name),
+            'drilldown': ('activity_programme_number.programme_number|'
+                          'activity_programme_number.programme'),
+            'order': 'activity_programme_number.programme_number:asc',
+            'pagesize': 30
+        }
+        url = ('https://openspending.org/api/3/cubes/'
+               'fb2fa9b200eb3e56facc4c287002fc4d:{}'
+               'aggregate/').format(resource[year.slug[:4]])
+        try:
```

Comment:
I suggest we don't handle this exception and let the programme crash so we don't hide errors which appear like missing data. I tend to work on the principle of only handling exceptions that I really need to - especially when interfacing with other systems - so that I could become aware of unexpected things that can lead to misleading or missing data.